service/s3/s3manager/batch.go

```diff
@@ -309,6 +323,77 @@ func (d *BatchDelete) Delete(ctx aws.Context, iter BatchDeleteIterator) error {
 	return nil
 }
 
+// deleteBatch will delete a batch of items in the objects parameters.
+func deleteBatch(d *BatchDelete, objects []BatchDeleteObject) []Error {
```

Comment:
I think this function can be significantly simplified if checking for field parity is done in `Delete` method.  This would prevent the need for duplicate logic to send the `DeleteObjects` API.